app/models/hyrax/batch_ingest/batch.rb

```diff
@@ -2,14 +2,22 @@
 
 module Hyrax::BatchIngest
   class Batch < ApplicationRecord
+    STATUSES = ['received', 'accepted', 'enqueued', 'running', 'completed', 'failed'].freeze
+
     has_many :batch_items
 
+    validates :status, inclusion: { in: STATUSES }
+
     def completed?
       batch_items.all? { |item| item.status == :success || item.status == :failed }
     end
 
     def admin_set
       @admin_set ||= AdminSet.find(admin_set_id) if admin_set_id
+    rescue ActiveFedora::ObjectNotFoundError
```

Comment:
My idea here was to allow the `Batch` model to behave predictably with respect to how associations normally work... i.e. if you have an association to another object, the association accessor will either return `nil` or the object. We don't have a test for that on the `Batch` model, but maybe we should.

Reply:
But.. now I'm just seeing that `nil` will be returned in the event of a present, but invalid `admin_set_id`. I'll change this to return `nil` if no `admin_set_id`, and then let the error bubble up otherwise.